sdk/typescript/runtime/runtime_bun.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"typescript-sdk/internal/dagger"
+	"typescript-sdk/tsdistconsts"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type BunRuntime struct {
+	sdkSourceDir      *dagger.Directory
+	introspectionJSON *dagger.File
+	cfg               *moduleConfig
+	ctr               *dagger.Container
+}
+
+func NewBunRuntime(
+	cfg *moduleConfig,
+	sdkSourceDir *dagger.Directory,
+	introspectionJSON *dagger.File,
+) *BunRuntime {
+	// Q: Should the cacheVolumeName depends on the cfg.image version
+	cacheVolumeName := fmt.Sprintf("mod-bun-cache-%s", tsdistconsts.DefaultBunVersion)
+
+	ctr := dag.Container().
+		From(cfg.image).
+		WithMountedCache("/root/.bun/install/cache", dag.CacheVolume(cacheVolumeName), dagger.ContainerWithMountedCacheOpts{
+			Sharing: dagger.CacheSharingModePrivate,
```

Comment:
Did you confirm `Shared` doesn't work here? Private means that if multiple containers try to use it in parallel it will be thrown out and a new one created. The performance will not be great

Reply:
How does it work if multiple module write data in the cache in parallel? If that is supported then I think we could move it to shared